lib/rex/payloads/meterpreter/config.rb

```diff
@@ -36,6 +36,9 @@ def is_x86?
   end
 
   def to_str(item, size)
+    if  (item.size > size-1)
+      item = item.slice(0..(size-1)) # TODO silently fix things or throw error?
```

Comment:
You should definitely check the size and warn the user about what the limit is. Silently fixing errors will only lead to pain and confusion for us down the road.

Reply:
Come to think of it, this is the wrong place to do the check. @busterb suggests checking the datastore option's length, and he is correct. That will also allow you to use `print_error` or otherwise report the problem. This is the wrong place to be printing, much less silently fixing the overflow (difficult to debug).

Reply:
Though if you're trying to fix the systemic problem instead of just `HttpUserAgent`, let's continue to discuss this. Either way, we should `raise` and/or `print_error`.